ctf/tests/core_suite.lua

```diff
@@ -0,0 +1,91 @@
+local lunatest = require("lunatest")
+--minetest environment
+vector = {}
+minetest = {}
+function minetest.after(val, fx)
+end
+
+--ctf environment
+ctf = {}
+
+-- to include relative path to core
+package.path = package.path .. ";../core.lua"
+require("../core")
+
+local assert_equal = lunatest.assert_equal
+local assert_not_nil = lunatest.assert_not_nil
+
+function assert_tables_equal(expected, actual)
+	for key,value in pairs(expected) do
+		assert_equal(expected[key], actual[key])
+	end
+end
+
+function assert_keysets_are_equal(sample, actual)
+	for key in pairs(sample) do
+		assert_not_nil(actual[key])
+	end
+	for key in pairs(actual) do
+		assert_not_nil(sample[key])
+	end
+end
+
+function test_shouldReturnPairsEquivalent()
+	local tbl = {}
+	tbl.b = 0
+	tbl.a = 1
+	tbl.c = 2
+	local nextFx, t, index = orderedPairs(tbl)
+	assert_equal("a", nextFx(t, index))
+	assert_equal("b", nextFx(t, "a"))
+	assert_equal("c", nextFx(t, "b"))
+	assert_equal(nil, nextFx(t, "c"))
+end
+
+function test_shouldReturnNilOnEmptyTable()
+	local tbl = {}
+	local nextFx, t, index = orderedPairs(tbl)
+	assert_equal(nil, nextFx(t, index))
+end
+
+function test_shouldBeAbleToIterate()
+	local tbl = {}
+	tbl.c = 0
+	tbl.a = 1
+	tbl.b = 2
+	local result = ""
+	for k,v in orderedPairs(tbl) do
+		result = result .. "(" .. k .. ";" .. v .. ")"
+	end
+	assert_equal("(a;1)(b;2)(c;0)", result)
+end
+
+function test_shouldAddOnlyIndexAndCleanup()
+	local tbl = {}
+	tbl.a = 0
+	tbl.c = 1
+	tbl.b = 2
+	local nextFx, t, index = orderedPairs(tbl)
+	local temp = nextFx(t,index)
+	assert_keysets_are_equal({a=0,c=0,b=0,__orderedIndex={}},t)
+	temp = nextFx(t,temp)
+	temp = nextFx(t,temp)
+	nextFx(t,temp)
+	assert_keysets_are_equal({a=0,c=0,b=0},t)
+end
+
+function test_shouldUseTableReferenceToIterate()
```

Comment:
The name is unclear, what exactly is this testing?
The ability to iterate using the return value (which I am assuming is the table reference) of orderedPairs?
The ability to update the value in the orderedPairs?
Imo, if it's both, it could use a better name or maybe it should be separated.